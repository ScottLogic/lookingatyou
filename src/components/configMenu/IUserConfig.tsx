export default interface IUserConfig {
    xSensitivity: number;
    ySensitivity: number;
    fps: number;
    swapEyes: boolean;
    toggleDebug: boolean;
    irisColor: string;
    bbox: number[];
}
