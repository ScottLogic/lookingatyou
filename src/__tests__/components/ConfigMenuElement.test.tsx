import { shallow } from 'enzyme';
import jsdom from 'jsdom';
import React from 'react';
import {
    ConfigMenuElement,
    ConfigMenuElementProps,
} from '../../components/configMenu/ConfigMenuElement';
import IUserConfig from '../../components/configMenu/IUserConfig';

let props: ConfigMenuElementProps;
let config: IUserConfig;
let window: Window;

describe('ConfigMenuElement tests', () => {
    beforeEach(() => {
        config = {
            xSensitivity: 1,
            ySensitivity: 1,
            fps: 30,
            swapEyes: false,
            toggleDebug: false,
            irisColor: 'blue',
        };
        window = new jsdom.JSDOM(`...`, { url: 'http://localhost' }).window;

        props = {
            config,
            window,
            setConfig: jest.fn(),
        };
    });

    it('should render correctly', () => {
        const wrapper = shallow(<ConfigMenuElement {...props} />).debug();
        expect(wrapper).toMatchSnapshot();
    });
});
