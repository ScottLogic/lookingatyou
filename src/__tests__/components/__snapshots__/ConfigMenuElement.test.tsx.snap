// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigMenuElement tests should render correctly 1`] = `
"<ConfigMenu width=\\"14em\\" timerLength={1000} window={{...}} debugEnabled={false}>
  <h1>
    Config
  </h1>
  <span className=\\"icon\\" data-tip={true} data-for=\\"APP\\">
    ?
  </span>
  <Memo() name=\\"FPS\\" configName=\\"fps\\" step={1} defaultValue={30} onValidInput={[undefined]} helpWith=\\"FPS\\" min={1} />
  <br />
  <Memo() name=\\"X Sensitivity\\" configName=\\"xSensitivity\\" step={0.1} defaultValue={1} onValidInput={[undefined]} helpWith=\\"X_SENSITIVITY\\" min={0} />
  <Memo() name=\\"Y Sensitivity\\" configName=\\"ySensitivity\\" step={0.1} defaultValue={1} onValidInput={[undefined]} helpWith=\\"Y_SENSITIVITY\\" min={0} />
  <Memo() name=\\"Iris Colour\\" configName=\\"irisColor\\" color=\\"blue\\" onInputChange={[undefined]} helpWith=\\"IRIS_COLOUR\\" />
  <br />
  <Memo() name=\\"Iris Colour\\" configName=\\"irisColor\\" color=\\"blue\\" onInputChange={[Function: mockConstructor]} helpWith=\\"IRIS_COLOUR\\" />
  <Memo() name=\\"Show Reflect\\" configName=\\"toggleReflection\\" helpWith=\\"REFLECTION\\" checked={[undefined]} onInputChange={[Function: mockConstructor]} />
  <Memo() name=\\"Toggle Debug\\" configName=\\"toggleDebug\\" helpWith=\\"DEBUG\\" checked={false} onInputChange={[Function: mockConstructor]} />
  <Help problemWith=\\"FPS\\" />
  <Help problemWith=\\"X_SENSITIVITY\\" />
  <Help problemWith=\\"Y_SENSITIVITY\\" />
  <Help problemWith=\\"VIDEO_STREAM\\" />
  <Help problemWith=\\"IRIS_COLOUR\\" />
  <Help problemWith=\\"APP\\" />
  <Help problemWith=\\"DEBUG\\" />
  <Help problemWith=\\"REFLECTION\\" />
  <Help problemWith=\\"REFLECTION_OPACITY\\" />
  <Help problemWith=\\"ARCHITECTURE\\" />
  <Help problemWith=\\"OUTPUT_STRIDE\\" />
  <Help problemWith=\\"RESOLUTION\\" />
  <Help problemWith=\\"MULTIPLIER\\" />
  <Help problemWith=\\"DETECTIONS\\" />
  <Help problemWith=\\"FLIP\\" />
  <Help problemWith=\\"MIN_SCORE\\" />
  <Help problemWith=\\"NMS_RADIUS\\" />
</ConfigMenu>"
`;
